src/dotty/tools/dotc/transform/SyntheticMethods.scala

```diff
@@ -118,9 +123,22 @@ class SyntheticMethods extends MiniPhaseTransform with IdentityDenotTransformer
 
     /** The class
      *
+     *  class C(x: T) extends AnyVal
+     *
+     *  gets the hashCode method:
+     *
+     *    def hashCode: Int = x.hashCode()
+     */
+    def valueHashCodeBody(implicit ctx: Context): Tree = {
+      assert(accessors.length == 1)
+      ref(accessors.head).select(nme.hashCode_).appliedToNone
```

Comment:
My assumption is that some of `hashcodes` you see are `ExprType`s and other ones are `MethodType`s. 
look for `ensureApplied` in codebase.